cmd/integrationArtifactDeploy.go

```diff
@@ -92,14 +94,135 @@ func runIntegrationArtifactDeploy(config *integrationArtifactDeployOptions, tele
 		log.Entry().
 			WithField("IntegrationFlowID", config.IntegrationFlowID).
 			Info("successfully deployed into CPI runtime")
-		return nil
+		error := PollIFlowDeploymentStatus(retryCount, config, httpClient)
+		return error
 	}
 	responseBody, readErr := ioutil.ReadAll(deployResp.Body)
 
 	if readErr != nil {
-		return errors.Wrapf(readErr, "HTTP response body could not be read, Response status code : %v", deployResp.StatusCode)
+		return errors.Wrapf(readErr, "HTTP response body could not be read, response status code: %v", deployResp.StatusCode)
 	}
-
 	log.Entry().Errorf("a HTTP error occurred! Response body: %v, Response status code : %v", responseBody, deployResp.StatusCode)
-	return errors.Errorf("Integration Flow deployment failed, Response Status code: %v", deployResp.StatusCode)
+	return errors.Errorf("integration flow deployment failed, response Status code: %v", deployResp.StatusCode)
+}
+
+//PollIFlowDeploymentStatus - Poll the integration flow deployment status,retrun status or error details
+func PollIFlowDeploymentStatus(retryCount int, config *integrationArtifactDeployOptions, httpClient piperhttp.Sender) error {
+
+	if retryCount <= 0 {
+		return errors.New("failed to start integration artifact after retrying several times")
+	}
+	deployStatus, err := GetIntegrationArtifactDeployStatus(config, httpClient)
+	if err != nil {
+		return err
+	}
+
+	//if artifact starting, then retry based on provided retry count
+	//with specific delay between each retry
+	if deployStatus == "STARTING" {
+		// Calling Sleep method
+		sleepTime := int(retryCount * 3)
+		time.Sleep(time.Duration(sleepTime) * time.Second)
+		retryCount--
+		return PollIFlowDeploymentStatus(retryCount, config, httpClient)
+	}
+
+	//if artifact started, then just return
+	if deployStatus == "STARTED" {
+		return nil
+	}
+
+	//if error return immediately with error details
+	if deployStatus == "ERROR" {
+		resp, err := GetIntegrationArtifactDeployError(config, httpClient)
+		if err != nil {
+			return err
+		}
+		return errors.New(resp)
+	}
+	return nil
+}
+
+//GetHTTPErrorMessage -Return HTTP failure message
+func GetHTTPErrorMessage(httpErr error, response *http.Response, httpMethod, statusURL string) (string, error) {
+	responseBody, readErr := ioutil.ReadAll(response.Body)
+	if readErr != nil {
+		return "", errors.Wrapf(readErr, "HTTP response body could not be read, response status code: %v", response.StatusCode)
+	}
+	log.Entry().Errorf("a HTTP error occurred! Response body: %v, response status code: %v", string(responseBody), response.StatusCode)
+	return "", errors.Wrapf(httpErr, "HTTP %v request to %v failed with error: %v", httpMethod, statusURL, responseBody)
+}
+
+//GetIntegrationArtifactDeployStatus - Get integration artifact Deploy Status
+func GetIntegrationArtifactDeployStatus(config *integrationArtifactDeployOptions, httpClient piperhttp.Sender) (string, error) {
```

Comment:
No need to be exported with capital letter, right?

Reply:
done